### PR TITLE
Set runtime check for discovery and p2p protocols

### PR DESCRIPTION
**Pull request by Acuadros95**

### CODE REVIEW

File: src/cpp/transport/Server.cpp [hunk 1]
```diff
@@ -152,7 +157,7 @@ template<typename EndPoint>
 bool Server<EndPoint>::enable_p2p(uint16_t p2p_port)
 {
     bool rv = false;
-    if (running_cond_)
+    if (has_p2p() && running_cond_)
```
Comment (Acuadros95) [inline src/cpp/transport/Server.cpp:160]: This check is not really necessary, as the Server class has a `return false` implementation for `init_discovery` and `init_p2p`

**Comment by Acuadros95** [comment]
https://github.com/eProsima/Micro-XRCE-DDS-Agent/blob/fa1ca7737272cd71851a53cb4b8c663cbb128758/include/uxr/agent/utils/ArgumentParser.hpp#L467-L472

Should we show an error message if the user tries to configure the discovery/p2p flags with an incompatible transport?

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Agent%20Manual%20Linux/443/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Agent%20Manual%20Windows/444/)

**Review by pablogs9 — APPROVED** [review]

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Agent%20Manual%20Linux/444/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Agent%20Manual%20Windows/445/)

**Pull request merged**